### PR TITLE
Enable alicloud

### PR DESCRIPTION
**Pull request by kevin-lacoo**
**What this PR does / why we need it**:
Alicloud support added.

**Release note**:
```feature user
Alicloud support added.
```

### CODE REVIEW

**Comment by vlerenc** [comment]
Uh, a PR since May... sorry, but why haven't you pinged me? I am subscribed to hundreds of repos and cannot watch everything/my Outlook rules only react if I am mentioned directly. I will take a look when I find the time, but next time, please ping me. :-)

File: chaosgarden/alicloud/actions.py [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+        logger.warning(f'tag network acl failed !')
+    
+    # delete block acl
+    if not REUSE_ACL:
+        alibot.delete_network_acl(AclId=blocking_acl_id)
+        logger.info(f'acl {blocking_acl_id} deleted.')
+    
+    logger.info(f'unblock {vpc_id} completed! ')
+    
+
```
Comment (vlerenc) [inline chaosgarden/alicloud/actions.py:264]: Maybe this block can be split up and moved to the specific sections like also with all the other providers? Or a separate section at the bottom (considering that these are helpers) like [here](https://github.com/gardener/chaos-engineering/blob/main/chaosgarden/garden/actions.py#L295)?

**Review by vlerenc — COMMENTED** [review]
I did only a superficial check. Looks good so far. Thank you very much.

I will take a deeper look, once you have set up the test clusters and can run the module against them.

The `__init__.py` contains significant amounts of code. Please make sure that no IP was violated and all code is yours and yours alone.

File: requirements.txt [hunk 2]
```diff
@@ -11,6 +11,9 @@ openstacksdk>=1.0.1,<2
 python-box>=7.0.0,<8
 PyYAML>=6.0,<7
 urllib3>=1.26.14,<2
+aliyun-python-sdk-core>=2.13.36
+aliyun-python-sdk-ecs>=4.24.30
+aliyun-python-sdk-vpc>=3.0.33
```
[the remark this answers is not in the document]
Comment (kevin-lacoo) [reply requirements.txt]: Upper bound added.

File: docs/alicloud/assess-filters-impact.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+                }
+            }
+        }
+    ],
+    "configuration": {
+        "ali_region": "world-1",                                                  # replace with your region;  field name fixed
+        "ali_zone": "world-1a",                                                   # replace with your zone;    field name free
+        "ali_filters": {                                                          # replace with your filters; field name free
+            "instances": {                                                         # optional instance filter;  field name fixed
+                "Tag-key": "kubernetes.io/cluster/shoot--project_name--shoot_name"        # replace with your kubernetes cluster tag
```
[the remark this answers is not in the document]
Comment (kevin-lacoo) [reply docs/alicloud/assess-filters-impact.json]: changed

File: docs/alicloud/run-compute-failure-simulation.json [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+                }
+            }
+        }
+    ],
+    "configuration": {
+        "ali_region": "world-1",                                                  # replace with your region;  field name fixed
+        "ali_zone": "world-1a",                                                   # replace with your zone;    field name free
+        "ali_filters": {                                                          # replace with your filters; field name free
+            "instances": {                                                         # optional instance filter;  field name fixed
+                "Tag-key": "kubernetes.io/cluster/shoot--project_name--shoot_name"        # replace with your kubernetes cluster tag
```
[the remark this answers is not in the document]
Comment (kevin-lacoo) [reply docs/alicloud/run-compute-failure-simulation.json]: changed

File: docs/alicloud/run-network-failure-simulation.json [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+                }
+            }
+        }
+    ],
+    "configuration": {
+        "ali_region": "world-1",                                                  # replace with your region;  field name fixed
+        "ali_zone": "world-1a",                                                   # replace with your zone;    field name free
+        "ali_filters": {                                                          # replace with your filters; field name free
+            "instances": {                                                         # optional instance filter;  field name fixed
+                "Tag-key": "kubernetes.io/cluster/shoot--project_name--shoot_name"        # replace with your kubernetes cluster tag
```
[the remark this answers is not in the document]
Comment (kevin-lacoo) [reply docs/alicloud/run-network-failure-simulation.json]: changed

File: chaosgarden/alicloud/actions.py (hunk 1 above)
Comment (kevin-lacoo) [reply chaosgarden/alicloud/actions.py]: Moved the code to Helper section.

**Comment by kevin-lacoo** [comment]
> I did only a superficial check. Looks good so far. Thank you very much.
> 
> I will take a deeper look, once you have set up the test clusters and can run the module against them.
> 
> The `__init__.py` contains significant amounts of code. Please make sure that no IP was violated and all code is yours and yours alone.

Yes, I wrote the entire __init__.py. Since Alibaba did not provide a suitable function packaging module, I wrote an AliyunBot class to wrap the Alibaba APIs that are used.

**Comment by vlerenc** [comment]
Some post-cleanups done:
- Added release note
- Commits not squashed, squashed now
- Editor removed some non-compliant (PEP8) spaces/new lines, but there are a ton more
- Replaced unused variables with `_` and dropped unused imports, too
- The deepcopy wasn't working @kevin-lacoo: `TypeError: cannot pickle 'module' object`; I removed the deepcopy code, please reintroduce, if you believe it is needed/should work, but it didn't for me
- I noticed a warning that should probably be debug message, no @kevin-lacoo?
  ```
  [WARNING] no exists alc for mode total in vpc vpc-gw8qv9kg4hi79ojv0rvye, unblock exited!!
  ```

**Review by vlerenc — APPROVED** [review]
Thank you @kevin-lacoo! I tested the main functionality and everything worked for me...

...with the notes above. Please check the `deepcopy` especially. I dropped it from your sources while testing, but it's still in there in git.

**Comment by kevin-lacoo** [comment]
> Thank you @kevin-lacoo! I tested the main functionality and everything worked for me...
> 
> ...with the notes above.
> 
> Please check the `deepcopy` especially. I dropped it from your sources while testing, but it's still in there in git.

Hi @vlerenc , my python version is 3.9 and I found the `deepcopy` call for Ali request object will fail in version 3.6. I don't know if the test env can use python 3.9 or later?

**Comment by vlerenc** [comment]
I have `3.11`.

Why is that needed? I removed the `deepcopy` and invoked the method twice (it's probably only relevant during retry). The second call worked just as fine. Are you sure, `deepcopy` is required?

**Pull request merged**